cdap-app-fabric/src/test/java/co/cask/cdap/app/runtime/monitor/RuntimeMonitorTest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.runtime.monitor;
+
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.cdap.api.dataset.lib.CloseableIterator;
+import co.cask.cdap.api.messaging.Message;
+import co.cask.cdap.api.messaging.MessageFetcher;
+import co.cask.cdap.api.messaging.MessagingContext;
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.client.config.ConnectionConfig;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.internal.app.runtime.messaging.MultiThreadMessagingContext;
+import co.cask.cdap.internal.app.runtime.monitor.MonitorMessage;
+import co.cask.cdap.internal.app.runtime.monitor.RuntimeMonitor;
+import co.cask.cdap.internal.guice.AppFabricTestModule;
+import co.cask.cdap.messaging.MessagingService;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import com.google.common.util.concurrent.Service;
+import com.google.gson.Gson;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.sun.net.httpserver.HttpExchange;
```

Comment:
Don't use these `com.sun` stuff. I think the unit-test only make sense if you have the server side of the API as well.

Reply:
I see. But we still need to test client side logic no? should I use `JettyServer` instead.

Reply:
?? When you have the server side code done, then you use that to test both server and client logic. Why need a jetty server?

Reply:
yea, but util we add server side code, should I use `com.sun`?

Reply:
I think you can keep the code for now, but have to change this test class and test methods to actually use the runtime monitor server.